csfieldguide/static/interactives/sorting-algorithms/js/sorting-algorithms.js

```diff
@@ -0,0 +1,170 @@
+    var drake = dragula([
+        document.getElementById('box-a'),
+        document.getElementById('box-b'),
+        document.getElementById('box-c'),
+        document.getElementById('box-d'),
+        document.getElementById('box-e'),
+        document.getElementById('box-f'),
+        document.getElementById('box-g'),
+        document.getElementById('box-h'),
+        document.getElementById('box-i'),
+        document.getElementById('box-j'),
+        document.getElementById('box-k'),
+        document.getElementById('box-l'),
+        document.getElementById('box-m'),
+        document.getElementById('box-n'),
+        document.getElementById('box-o'),
+        document.getElementById('box-p'),
+        document.getElementById('box-q'),
+        document.getElementById('box-r'),
+        document.getElementById('box-s'),
+        document.getElementById('box-t'),
+        document.getElementById('box-u'),
+        document.getElementById('box-v'),
+        document.getElementById('box-w'),
+        document.getElementById('box-x'),
+        document.getElementById('box-y'),
+        document.getElementById('box-z')
+    ]);
+
+    drake.on('drop', (target, source) => {
+        if (source.children.length == 2) { // means an element has been dropped in this div 
+            swap(target, source);
+        }
+        compareWeights();
+    });
+});
+
+function swap(target, source) {
+    // get id of moved image
+    var moved_img_id = target.id;
+    
+    // get the div that the moved image came from
+    var target_div_id = moved_img_id.slice(0, 5);
+    var target = document.getElementById(target_div_id);
+    
+    // figure out which image needs to be moved
+    var img_to_move;
+    if (source.children[0].id != moved_img_id) {
+        img_to_move = source.children[0];
+    } else if (source.children[1].id != moved_img_id) {
+        img_to_move = source.children[1];
+    }
+
+    target.appendChild(img_to_move);
+
+    var tmp_id = target.id;
+    target.id = source.id;
+    source.id = tmp_id;
+}
+
+function compareWeights() {
+    var left_weight_div = document.getElementsByClassName('left-weight-content')[0];
+    var right_weight_div = document.getElementsByClassName('right-weight-content')[0];
+    var left_weight = left_weight_div.children[0].dataset.weight;
+    var right_weight = right_weight_div.children[0].dataset.weight;
+
+    // check if both are placeholder images
+    if (left_weight == 0 && right_weight == 0) {
+        rotateIndicator('middle');
+    } else {
+        if (left_weight > right_weight) { // left is heavier
+            rotateIndicator('left');
+        } else if (right_weight > left_weight) { // right is heavier
+            rotateIndicator('right');
+        }
+        if (left_weight != 0 && right_weight != 0) {
+            countComparisons();
+        }
+    }
+}
+
+function rotateIndicator(direction) {
+    indicator = document.getElementById('scale');
+    if (direction == 'left') {
```

Comment:
Instead of having to repeat all the different rules, you could add a CSS class defined in a SCSS file, which just has the `transform` rule. Because when we compile SCSS it automatically creates the rules for all browsers.

Reply:
I love this idea! I see that some other interactives have an scss folder, but I can't work out how/when it is being compiled. What am I missing? (I tried just making an scss folder in my interactive and running update_static to see if it's just magic, but it is not)

Reply:
Why can't I submit this?

Reply:
That sounds correct for it to work, could you please show me your setup for this?